test_runner/regress/test_storage_scrubber.py

```diff
@@ -188,6 +189,7 @@ def test_scrubber_physical_gc_ancestors(
             # Small layers and low compaction thresholds, so that when we split we can expect some to
             # be dropped by child shards
             "checkpoint_distance": f"{1024 * 1024}",
+            "checkpoint_timeout": "5s",
```

Comment:
why do you add this? is it to test the timeout? I don't see any sleep for 5 seconds in the code though.

Reply:
The intent is to ensure prompt uploads without having to call /checkpoint, although there were several failures on this PR so something isn't right...

Reply:
This wasn't working, because low values of checkpoint_timeout aren't enforced -- layer rolling happens on a relaxed timescale via the task that runs `ingest_housekeeping`.

I re-worked this change to be simpler, by just making compaction in checkpoints optional, so that the test can do what it wants: checkpointing without a compaction.